plugins/inputs/jolokia2/jolokia.go

```diff
@@ -0,0 +1,220 @@
+
+	Metrics []MetricConfig `toml:"metric"`
+}
+
+func (j *Jolokia) SampleConfig() string {
+	return `
+  # default_tag_prefix      = ""
+  # default_field_prefix    = ""
+  # default_field_separator = "."
+
+  ## Add agents to query
+  agents = ["http://localhost:8080/jolokia"]
+  # username = ""
+  # password = ""
+  # response_timeout = "5s"
+
+  ## Optional SSL config
+  # ssl_ca   = "/var/private/ca.pem"
+  # ssl_cert = "/var/private/client.pem"
+  # ssl_key  = "/var/private/client-key.pem"
+  # insecure_skip_verify = false
+
+  ## Add metrics to read
+  [[inputs.jolokia2.metric]]
+    name  = "java_runtime"
+    mbean = "java.lang:type=Runtime"
+    paths = ["Uptime"]
+`
+}
+
+func (j *Jolokia) Description() string {
+	return "Read JMX metrics from a Jolokia REST endpoint"
+}
+
+func (j *Jolokia) Gather(acc telegraf.Accumulator) error {
+	var metrics []Metric
+
+	for _, config := range j.Metrics {
+		metrics = append(metrics, NewMetric(config,
+			j.DefaultFieldPrefix, j.DefaultFieldSeparator, j.DefaultTagPrefix))
+	}
+
+	gatherer := NewGatherer(acc, metrics)
+	for _, url := range j.URLs {
+		agent, err := j.buildAgent(url)
+		if err != nil {
+			return err
+		}
+
+		err = gatherer.Gather(agent)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (j *Jolokia) buildAgent(url string) (*Agent, error) {
+	return NewAgent(url, &AgentConfig{
+		Username:           j.Username,
+		Password:           j.Password,
+		ResponseTimeout:    j.ResponseTimeout,
+		SSLCA:              j.SSLCA,
+		SSLCert:            j.SSLCert,
+		SSLKey:             j.SSLKey,
+		InsecureSkipVerify: j.InsecureSkipVerify,
+	})
+}
+
+type JolokiaProxy struct {
+	DefaultFieldPrefix    string
+	DefaultFieldSeparator string
+	DefaultTagPrefix      string
+
+	URL                   string `toml:"url"`
+	DefaultTargetPassword string
+	DefaultTargetUsername string
+	Targets               []JolokiaProxyTargetConfig `toml:"target"`
+
+	Username           string
+	Password           string
+	SSLCA              string `toml:"ssl_ca"`
+	SSLCert            string `toml:"ssl_cert"`
+	SSLKey             string `toml:"ssl_key"`
+	InsecureSkipVerify bool
+	ResponseTimeout    time.Duration `toml:"response_timeout"`
+
+	Metrics []MetricConfig `toml:"metric"`
+}
+
+type JolokiaProxyTargetConfig struct {
+	URL      string `toml:"url"`
+	Username string
+	Password string
+}
+
+func (jp *JolokiaProxy) SampleConfig() string {
+	return `
+  # default_tag_prefix      = ""
+  # default_field_prefix    = ""
+  # default_field_separator = "."
+
+  ## Proxy agent
+  url = "http://localhost:8080/jolokia"
+  # username = ""
+  # password = ""
+  # response_timeout = "5s"
+
+  ## Optional SSL config
+  # ssl_ca   = "/var/private/ca.pem"
+  # ssl_cert = "/var/private/client.pem"
+  # ssl_key  = "/var/private/client-key.pem"
+  # insecure_skip_verify = false
+
+  ## Add targets to query
+  # default_target_username = ""
+  # default_target_password = ""
+  [[inputs.jolokia_proxy.target]]
```

Comment:
Yes, the proxy can make requests to multiple targets, and each of those targets could potentially have different user/pass. This would be in a world where, for example, a distributed application's JMX endpoints are completely inaccessible except to the proxy. The proxy can have its own user/pass, and the individual targets can have their own user/pass. It's quite difficult to express all these possibilities in a single table.